src/test/scala/vsys/blockchain/state/contract/channel/ChannelContractOpcDiffTest.scala

```diff
@@ -0,0 +1,96 @@
+  } yield (genesis, create)
+
+  property("register contract transaction pass OpcFunDiff"){
+    forAll(preconditionsAndRegContract) { case (genesis, create) =>
+      assertOpcFuncDifferEi(2, None, create) { opcFunDiffEi =>
+        opcFunDiffEi shouldBe an[Right[_, _]]
+      }
+      assertDiffEi(Seq(TestBlock.create(Seq(genesis))), TestBlock.createWithTxStatus(Seq(create), TransactionStatus.Success)) { blockDiffEi =>
+        blockDiffEi shouldBe an[Right[_, _]]
+      }
+    }
+  }
+
+  val regWrongParaContract: Gen[Contract] =
+    Gen.const(Contract.buildContract(ContractPaymentChannel.contract.languageCode, ContractPaymentChannel.contract.languageVersion,
+      wrongDataTrigger, ContractPaymentChannel.contract.descriptor, ContractPaymentChannel.contract.stateVar,
+      ContractPaymentChannel.contract.stateMap, ContractPaymentChannel.contract.textual).explicitGet())
+  val preconditionsAndRegContractWrongPara: Gen[RegisterContractTransaction] = for {
+    (master, ts, fee) <- ContractGenHelper.basicContractTestGen()
+    contract <- regWrongParaContract
+    sysTokenId = tokenIdFromBytes(ContractAccount.systemContractId.bytes.arr, Ints.toByteArray(0)).explicitGet()
+    data <- initPaymentChannelContractDataStackGen(sysTokenId.arr)
+    description <- validDescStringGen
+    create <- registerPaymentChannelGen(master, contract, data, description, fee, ts)
+  } yield create
+
+  property("payment channel contract transaction cannot pass due to wrong list of parameters"){
+    forAll(preconditionsAndRegContractWrongPara) { case create =>
+      assertOpcFuncDifferEi(2, None, create) { opcFunDiffEi =>
+        opcFunDiffEi shouldBe Left(ContractDataTypeMismatch)
+      }
+    }
+  }
+
+  val regWrongOpcFunContract: Gen[Contract] =
+    Gen.const(Contract.buildContract(ContractPaymentChannel.contract.languageCode, ContractPaymentChannel.contract.languageVersion,
+      wrongOpcTrigger, ContractPaymentChannel.contract.descriptor, ContractPaymentChannel.contract.stateVar,
+      ContractPaymentChannel.contract.stateMap, ContractPaymentChannel.contract.textual).explicitGet())
+  val preconditionsAndRegContractWrongFun: Gen[RegisterContractTransaction] = for {
+    (master, ts, fee) <- ContractGenHelper.basicContractTestGen()
+    contract <- regWrongOpcFunContract
+    sysTokenId = tokenIdFromBytes(ContractAccount.systemContractId.bytes.arr, Ints.toByteArray(0)).explicitGet()
+    data <- initPaymentChannelContractDataStackGen(sysTokenId.arr)
+    description <- validDescStringGen
+    create <- registerPaymentChannelGen(master, contract, data, description, fee, ts)
+  } yield create
+
+  property("register contract transaction cannot pass due to wrong opcode"){
+    forAll(preconditionsAndRegContractWrongFun) { case create =>
+      assertOpcFuncDifferEi(2, None, create) { opcFunDiffEi =>
+        opcFunDiffEi shouldBe Left(ContractInvalidOPCData)
+      }
+    }
+  }
+}
```

Comment:
It looks fine here. But it only tests the trigger part, may design fun part for more interactions.